enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,437 @@
+  - "@knobunc" # Staff engineer for Networking, involved in past discussions.
+approvers:
+  - "@JoelSpeed"
+  - "@knobunc"
+api-approvers:
+  - None
+creation-date: 2025-01-22
+last-updated: 2025-01-27
+tracking-link:
+  - https://issues.redhat.com/browse/NE-1946
+see-also:
+  - "/enhancements/ingress/gateway-api-with-cluster-ingress-operator.md"
+---
+
+
+# Gateway API CRD Life-Cycle Management
+
+This enhancement describes how the [Cluster Ingress Operator (CIO)] manages
+Gateway API Custom Resource Definitions (CRDs).
+
+[Cluster Ingress Operator (CIO)]:https://github.com/openshift/cluster-ingress-operator
+
+## Summary
+
+A key goal of Gateway API is to nurture a diverse ecosystem of implementations.
+These implementations, including OpenShift's, all rely on the same Gateway API
+CRDs. As such, OpenShift must manage the life-cycle of these CRDs in a way that
+enables Red Hat to support the product, minimizes friction for the end-user,
+provides consistency, and enables other implementations that rely on these
+CRDs.
+
+## Motivation
+
+Gateway API is an evolving API, and its CRDs change over time.  For core
+OpenShift, we must ensure that we have a version of the CRDs that is compatible
+with our product, and we must handle upgrading these CRDs for our own needs.  We
+must also avoid unnecessarily causing friction for Red Hat layered products,
+partner products, and other third-party implementations.
+
+### User Stories
+
+#### Using OpenShift's Gateway API implementation on a new OpenShift cluster
+
+As a cluster-admin, I want to install a new OpenShift 4.19 cluster, and then use
+OpenShift's Gateway API implementation to configure ingress to a workload on
+this cluster without having to deploy and then manage the lifecycle of any CRDs
+thereafter.
+
+#### Using OpenShift's Gateway API implementation on an upgraded cluster
+
+As a cluster-admin, I want to upgrade my cluster from OpenShift 4.18 (which
+*doesn't* manage the Gateway API CRDs' life-cycle) to OpenShift 4.19 (which
+*does*). If a user on my cluster or I had previously installed Gateway API
+CRDs and was managing their lifecycle, then I need OpenShift to explicitly
+confirm consent from me to take over the management of these CRDs to avoid
+disruptions to existing workloads.
+
+#### Using a third-party Gateway API implementation
+
+As a cluster-admin, I want to install a third-party Gateway API implementation
+on my OpenShift 4.19 cluster, and use the third-party implementation without
+any interference from the first-party implementation. Relatedly, I want to be
+able to utilize both the first-party and any third-party solution alongside
+each other simultaneously and independently without any interference between the
+two.
+
+#### Future OpenShift upgrades
+
+As a cluster-admin, I want to be able to receive updates to the Gateway API
+resources via zstream (and major) releases to add new features and capabilities.
+
+### Goals
+
```

Comment:
Probably the first goal should be:

Establish the owner of Gateway API CRDs as OpenShift Cluster Ingress Operator.

Reply:
```suggestion
* Establish the owner of Gateway API CRDs as OpenShift Cluster Ingress Operator.
* Prevent upgrades on OpenShift Clusters that have non-matching Gateway API CRDs already installed.
```